validation/hostname.go

```diff
@@ -1,17 +1,45 @@
 package main
 
 import (
+	"fmt"
+	"runtime"
+
+	"github.com/mndrix/tap-go"
 	"github.com/opencontainers/runtime-tools/validation/util"
 )
 
-func main() {
+func testHostname(t *tap.T, setHostname bool) error {
 	g, err := util.GetDefaultGenerator()
 	if err != nil {
-		util.Fatal(err)
+		return err
+	}
+
+	if setHostname {
+		g.SetHostname("hostname-specific")
 	}
-	g.SetHostname("hostname-specific")
 	err = util.RuntimeInsideValidate(g, nil)
+	t.Ok(err == nil, "hostname is set correctly")
 	if err != nil {
+		t.Diagnosticf("expect: err == nil, actual: err != nil")
+	}
+
+	return err
```

Comment:
We should return nil.
util.Fatal is used when an unexpected error detect, most caused by runtime-tools source code.

Reply:
I think we need a patch to make RuntimeInsideValidate and RuntimeOutsideValidate stronger.
Now it does not different runtime error and runtime-tool/system error.